service/frontend/middleware/global.go

```diff
@@ -12,12 +13,16 @@ func SetupGlobalMiddleware(handler http.Handler) http.Handler {
 	next = middleware.Logger(next)
 	next = middleware.Recoverer(next)
 
+	if authToken != nil {
+		next = TokenAuth("bearer", authToken.Token)(next)
+		basicAuth = nil
```

Comment:
I think that we don't need to set `basicAuth = nil` here which disable the basic auth for all request. This is because we can skip the basicAuth middleware on HTTP request basis. When an API token is set on a request, we can skip basic authentication for the specific request only. As I put in the comment below, let's proceed without considering how to skip basic auth middleware.